### PR TITLE
Add resource_type in AddInstanceResponse

### PR DESCRIPTION
**Pull request by site0801**

### CODE REVIEW

File: pkg/starter/starter.go [hunk 1]
```diff
@@ -192,7 +192,7 @@ func (s *Starter) processJob(ctx context.Context, job datastore.Job) error {
 		IPAddress:      ipAddress,
 		TargetID:       job.TargetID,
 		CloudID:        cloudID,
-		ResourceType:   target.ResourceType,
+		ResourceType:   resourceType,
```
Comment (whywaita) [inline pkg/starter/starter.go:195]: `resoruceType` response is **optional** for backward compatibility.

You need to check that `resourceType` is nil. Please set `target.ResourceType` if nil.

**Review by whywaita — CHANGES_REQUESTED** [review]

File: pkg/starter/starter.go (hunk 1 above)
Comment (site0801) [reply pkg/starter/starter.go:198]: If `resoruceType` response is zero (= unknown), then reference the value of `target.ResourceType` (I think my way is not good).
https://github.com/whywaita/myshoes/pull/169/commits/20ce141cf376298b5a9aec23a2f99186115f725b#diff-3a0c64b483dd2fed243139d0725e8be15f12e642cd167cd9ce85923e63d138c5R170-R172

Also, there was an omission in the fix for shoes.go, so I fixed it.
https://github.com/whywaita/myshoes/pull/169/commits/20ce141cf376298b5a9aec23a2f99186115f725b#diff-e1765ca0db77f2dddd1d22c7afe51ed99e53823c487b61eefd75950772097778R82-R94

File: pkg/starter/starter.go [hunk 2]
```diff
@@ -170,6 +170,9 @@ func (s *Starter) processJob(ctx context.Context, job datastore.Job) error {
 
 		return fmt.Errorf("failed to bung (target ID: %s, job ID: %s): %w", job.TargetID, job.UUID, err)
 	}
+	if resourceType == 0 {
```
Comment (whywaita) [inline pkg/starter/starter.go:173]: Please use `Unknown` in package

Comment (site0801) [reply pkg/starter/starter.go]: https://github.com/whywaita/myshoes/pull/169/commits/acbbb5ba6437dd4f154bdde2e3a3b1ea044fa550
Fixed in this commit.

File: pkg/shoes/shoes.go [hunk 3]
```diff
@@ -88,10 +88,10 @@ func (c *GRPCClient) AddInstance(ctx context.Context, runnerName, setupScript st
 	}
 	resp, err := c.client.AddInstance(ctx, req)
 	if err != nil {
-		return "", "", "", fmt.Errorf("failed to AddInstance: %w", err)
+		return "", "", "", datastore.ResourceType(0), fmt.Errorf("failed to AddInstance: %w", err)
```
Comment (whywaita) [inline pkg/shoes/shoes.go:91]: Please use `datastore.ResourceTypeUnknown`

File: pkg/shoes/shoes.go [hunk 4]
```diff
@@ -88,10 +88,10 @@ func (c *GRPCClient) AddInstance(ctx context.Context, runnerName, setupScript st
...
 	}
 
-	return resp.CloudId, resp.IpAddress, resp.ShoesType, nil
+	return resp.CloudId, resp.IpAddress, resp.ShoesType, datastore.ResourceType(resp.ResourceType), nil
```
Comment (whywaita) [inline pkg/shoes/shoes.go:94]: Please use unmarshal command https://github.com/whywaita/myshoes/blob/8d9faf2330a46727c6b43ddd6274424f537edd6c/pkg/datastore/resource_type.go#L84

**Review by whywaita — CHANGES_REQUESTED** [review]

File: pkg/shoes/shoes.go (hunk 3 above)
Comment (site0801) [reply pkg/shoes/shoes.go]: Fixed in https://github.com/whywaita/myshoes/pull/169/commits/5a641811fe6c4bd8e4c9ac8b62dc3991bb279e2d

File: pkg/shoes/shoes.go (hunk 4 above)
Comment (site0801) [reply pkg/shoes/shoes.go]: Fixed in https://github.com/whywaita/myshoes/pull/169/commits/5a641811fe6c4bd8e4c9ac8b62dc3991bb279e2d

**Pull request merged**